docs/misc/config.md

```diff
@@ -0,0 +1,95 @@
+Configuration
+=============
+
+Configurations define settings and consumer preferences that can be applied to a mod instance. Forge uses a configuration system using [TOML][toml] files and read with [night-config][nightconfig].
+
+Creating a Configuration
+------------------------
+
+A configuration can be created using a subtype of `IConfigSpec`. Forge implements the type via `ForgeConfigSpec` and enables its construction through `ForgeConfigSpec#Builder`. The builder can separate the config values into sections via `Builder#push` to create a section and `Builder#pop` to leave a section. Afterwards, the configuration can be built using one of two methods:
+
+Method    | Description
+:---      | :---
+build     | Creates the `ForgeConfigSpec`.
+configure | Creates a pair of the class holding the config values and the `ForgeConfigSpec`.
+
+Each config value can be supplied with additional context to provide additional behavior. Contexts must be defined before the config value is fully built:
+
+Method       | Description
+:---         | :---
+comment      | Provides a description of what the config value does. Can provide multiple strings for a multiline comment.
+translation  | Provides a translation key for the name of the config value.
+worldRestart | The world must be restarted before the config value can be changed.
+
+### ConfigValue
+
+Config values can be built with the provided contexts (if defined) using any of the `#define` methods. The methods take in four components: a path representing the name of the variable, the default value when no valid configuration is present, a validator to make sure the deserialized object is valid, and a class representing the data type of the config value.
+
+!!! note
+    The path is a `.` separated string representing the sections the config value is in.
```

Comment:
Perhaps it may be better to make these four components as an unordered list, and inlining the note about the path in the list.